File: web/vue/src/ai/api/pluginConfig.ts
/**
 * 插件配置管理 API
 *
 * 提供插件配置、测试连接、启动、停止等操作接口。
 * 对应后端 plugin_config_controller，路由前缀：/ai/console/v1/plugins/installations
 */

import { rawPost } from "@/framework/api/client";
import type { ApiResponse } from "@/framework/api/types";

// ==================== 类型定义 ====================

/**
 * 配置插件请求
 */
export interface ConfigPluginRequest {
  /** 插件能力配置，如 API Key、Endpoint 等 */
  plugin_config?: Record<string, unknown>;
  /** 运行时配置，如超时时间、重试次数等 */
  runtime_config?: Record<string, unknown>;
}

/**
 * 配置插件响应
 */
export interface ConfigPluginResponse {
  /** 插件 ID */
  plugin_id: string;
  /** 配置验证状态：null=未测试, true=验证通过, false=验证失败 */
  validated: boolean | null;
  /** 配置保存或验证的消息 */
  message?: string;
}

/**
 * 测试插件配置连接响应
 */
export interface TestPluginResponse {
  /** 插件 ID */
  plugin_id: string;
  /** 配置验证结果 */
  validated: boolean;
  /** 验证结果消息 */
  message: string;
}

/**
 * 启动插件响应
 */
export interface StartPluginResponse {
  /** 插件 ID */
  plugin_id: string;
  /** 插件状态：ACTIVE */
  status: string;
  /** 运行端口 */
  port?: number | null;
  /** 警告信息（如配置未验证） */
  warning?: string | null;
}

/**
 * 停止插件响应
 */
export interface StopPluginResponse {
  /** 插件 ID */
  plugin_id: string;
  /** 插件状态：INACTIVE */
  status: string;
}

// ==================== API 函数 ====================

const CONSOLE_BASE = "/ai/console/v1/plugins/installations";

/**
 * 配置插件
 *
 * 保存插件的能力配置和运行时配置
 */
export async function configPlugin(
  pluginId: string,
  data: ConfigPluginRequest,
): Promise<ApiResponse<ConfigPluginResponse>> {
  return rawPost(`${CONSOLE_BASE}/${encodeURIComponent(pluginId)}/config`, data);
}

/**
 * 测试插件配置连接
 *
 * 使用已保存的配置测试插件是否能正常连接
 */
export async function testPlugin(
  pluginId: string,
): Promise<ApiResponse<TestPluginResponse>> {
  return rawPost(`${CONSOLE_BASE}/${encodeURIComponent(pluginId)}/test`);
}

/**
 * 启动插件
 *
 * 启动状态为 INACTIVE 的插件，创建插件进程并更新状态为 ACTIVE
 */
export async function startPlugin(
  pluginId: string,
): Promise<ApiResponse<StartPluginResponse>> {
  return rawPost(`${CONSOLE_BASE}/${encodeURIComponent(pluginId)}/start`);
}

/**
 * 停止插件
 *
 * 停止状态为 ACTIVE 的插件，终止插件进程并更新状态为 INACTIVE
 */
export async function stopPlugin(
  pluginId: string,
): Promise<ApiResponse<StopPluginResponse>> {
  return rawPost(`${CONSOLE_BASE}/${encodeURIComponent(pluginId)}/stop`);
}

File: server/python/src/ai/schemas/plugin_config.py
"""插件配置相关 Schema

提供插件配置、测试、启动、停止等操作的请求和响应模型。

注意：这些 Schema 用于新的插件配置接口（路径参数版本），与现有的接口（Query 参数版本）分离。
"""

from __future__ import annotations

from framework.schemas import BaseModel


class PluginConfigRequest(BaseModel):
    """插件配置请求"""

    plugin_config: dict | None = None
    """插件能力配置，如 API Key、Endpoint 等"""

    runtime_config: dict | None = None
    """运行时配置，如超时时间、重试次数等"""


class PluginConfigResponse(BaseModel):
    """插件配置响应"""

    plugin_id: str
    """插件 ID"""

    validated: bool | None = None
    """配置验证状态：null=未测试, true=验证通过, false=验证失败"""

    message: str | None = None
    """配置保存或验证的消息"""


class PluginTestRequest(BaseModel):
    """插件测试请求"""

    pass  # 无需额外参数，使用已保存的配置测试


class PluginTestResponse(BaseModel):
    """插件测试响应"""

    plugin_id: str
    """插件 ID"""

    validated: bool
    """配置验证结果"""

    message: str
    """验证结果消息"""


class PluginStartRequest(BaseModel):
    """插件启动请求"""

    pass  # 无需额外参数


class PluginStartResponse(BaseModel):
    """插件启动响应"""

    plugin_id: str
    """插件 ID"""

    status: str
    """插件状态：ACTIVE"""

    port: int | None = None
    """运行端口"""

    warning: str | None = None
    """警告信息（如配置未验证）"""


class PluginStopRequest(BaseModel):
    """插件停止请求"""

    pass  # 无需额外参数


class PluginStopResponse(BaseModel):
    """插件停止响应"""

    plugin_id: str
    """插件 ID"""

    status: str
    """插件状态：INACTIVE"""
